packages/dcos-integration-test/extra/test_meta.py

```diff
@@ -0,0 +1,82 @@
+"""
+Tests for the integration test suite itself.
+"""
+
+import subprocess
+from pathlib import Path
+from typing import Set
+
+import yaml
+
+from get_test_group import patterns_from_group
+
+
+def _tests_from_pattern(ci_pattern: str) -> Set[str]:
+    """
+    From a CI pattern, get all tests ``pytest`` would collect.
+    """
+    tests = set([])  # type: Set[str]
+    args = ['pytest', '--collect-only', ci_pattern, '-q']
+    result = subprocess.run(args=args, stdout=subprocess.PIPE)
+    output = result.stdout
+    for line in output.splitlines():
+        if b'error in' in line:
+            message = (
+                'Error collecting tests for pattern "{ci_pattern}". '
+                'Full output:\n'
+                '{output}'
+            ).format(
+                ci_pattern=ci_pattern,
+                output=output,
+            )
+            raise Exception(message)
+        if (
+            line and
+            b'PytestWarning' not in line and
+            # Some tests are skipped on collection.
+            b'skipped in' not in line and
+            # Some tests are deselected by the ``pytest.ini`` configuration.
+            b' deselected' not in line and
```

Comment:
whitespace important?

Reply:
Yes 😞 

I'm not happy about all this string parsing but I am unaware of a different way.

I'll add a comment but basically if we don't have a space, we could ignore e.g. `test_secret_deselected` (hypothetical test name).